common/djangoapps/student/models.py

```diff
@@ -215,6 +215,29 @@ def is_username_retired(username):
     return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
 
 
+def is_email_retired(email):
+    """
+    Checks to see if the given email has been previously retired
+    """
+    locally_hashed_emails = user_util.get_all_retired_emails(
+        email,
+        settings.RETIRED_USER_SALTS,
+        settings.RETIRED_EMAIL_FMT
+    )
+
+    return User.objects.filter(email__in=list(locally_hashed_emails)).exists()
+
+
+def username_or_email_exists_or_retired(user_identifier):
+    """
+    Check a username or email against the User model for existence.
+    """
+    if '@' in user_identifier:
```

Comment:
Unfortunately I don't think this will work as we allow @ in usernames. Perhaps make separate username and email params to be explicit?
```
mysql> select count(id) from auth_user where username like '%@%';
+-----------+
| count(id) |
+-----------+
|       120 |
+-----------+
```

Reply:
I agree - this check seems broken. I know that this code is what existed before - but I don't see why we can't have both `email` and `username` optional parameters to this method and check the proper field/fields depending on which one the identifier is.